### PR TITLE
Add Makefile for LLVM builds

### LINKED ISSUE #22: Support for LLVM builds
As of now, LLVM Flang (`flang-new`) is able to build ASiMoV-CCS and its dependencies (though some require patches).

Would a Makefile that enables LLVM builds be welcomed?

### PR DESCRIPTION
**Pull request by rj-jesus**
Closes #22

### CODE REVIEW

File: build_tools/archs/Makefile.llvm [hunk 1]
```diff
@@ -0,0 +1,36 @@
+CC = mpicc
+CFLAGS = -O3
+
+FC = mpif90
+FFLAGS = -cpp -std=f2018 -fimplicit-none
+# HLFIR is required at least since 7411a59001ca7c77ca374cba2f986dfe614c0462
```
Comment (pbartholomew08) [inline build_tools/archs/Makefile.llvm:6]: I'm not very familiar with the development of Flang - is the commit a particular flang or LLVM commit? And is HLFIR an extension?

Could you perhaps add a (brief!) subsection in the build system (perhaps under where you've noted the architecture-specific flags) about the verion(s) of flang that can build CCS?

File: build_tools/archs/Makefile.llvm [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  UNAME_M := $(shell uname -m)
+  ifeq ($(UNAME_M),aarch64)
+    FFLAGS += -O3 -mcpu=native
+  else
+    FFLAGS += -O3 -march=native
+  endif
+endif
+ifeq ($(PROFILE),yes)
+  # Can be visualised with `<LLVM install dir>/share/opt-viewer/opt-viewer.py`
+  FFLAGS += -fsave-optimization-record
```
Comment (pbartholomew08) [inline build_tools/archs/Makefile.llvm:32]: This sounds cool!

**Review by pbartholomew08 — CHANGES_REQUESTED** [review]

File: build_tools/archs/Makefile.llvm (hunk 2 above)
Comment (pbartholomew08) [reply build_tools/archs/Makefile.llvm:32]: N.B. just a comment - no actual change requested here

File: build_tools/archs/Makefile.llvm (hunk 1 above)
Comment (rj-jesus) [reply build_tools/archs/Makefile.llvm:6]: Oh I should have been explicit, it's actually a [commit from CCS](https://github.com/asimovpp/asimov-ccs/commit/7411a59001ca7c77ca374cba2f986dfe614c0462).

HLFIR (High-Level Fortran IR) is another way of lowering Fortran code (it sits on top of FIR—Fortran IR). It's meant to be made the default option at some point, but for now the default is to lower Fortran to FIR directly.

I took note of the commit that made me enable HLFIR lowering as this caused a performance regression that might be worth looking into later.

> Could you perhaps add a (brief) subsection in the build system (perhaps under where you've noted the architecture-specific flags) about the verion(s) of flang that can build CCS?

Yep, will do. Not sure if any tagged release can build CCS currently, but I'll check!

**Pull request closed without merging**